Auto/IR/SMT.lean

```diff
@@ -277,6 +280,17 @@ def SMTOption.toString : SMTOption → String
 instance : ToString SMTOption where
   toString := SMTOption.toString
 
+def SMTReservedWords : HashSet String :=
+  let reserved := #[ "as", "let", "exists", "forall", "match", "par",
+    "assert", "check-sat", "check-sat-assuming",
+    "declare-const", "declare-datatype", "declare-datatypes",
+    "declare-fun", "declare-sort", "define-fun", "define-fun-rec", "define-funs-rec",
+    "define-sort", "echo", "exit", "get-assertions", "get-info",
+    "get-model", "get-option", "get-proof", "get-unsat-assumptions",
+    "get-unsat-core", "get-value", "pop", "push", "reset", "reset-assertions",
+    "set-info", "set-logic", "set-option"]
+  reserved.foldl (fun hs s => hs.insert s) {}
```

Comment:
Not sufficient. Pre-defined symbols in SMT-lib like ``+, -, *, /, %, ...`` and auxiliary definitions in Lean-auto like `iemod` should be excluded. Individual SMT solvers also have pre-defined symbols.

Reply:
Already causing issue in
``
example : (∀ (xs ys zs : List α), xs ++ ys ++ zs = xs ++ (ys ++ zs)) := by
  auto [List.append_assoc]
``
after previous pretty-print commit.

Reply:
![image](https://github.com/leanprover-community/lean-auto/assets/58621828/dbdbdde2-538f-49f5-809d-686535765018)

Reply:
@dranov The reserved words also contain numeral, decimal and string, but there is no guarantee that Lean's pretty printer won't print them. I'm feeling that using absolutely no predetermined prefix (previously ``smti_``) for Lean-auto generated names is causing too much trouble. Can we maybe add something like ``_`` as a prefix?